### PR TITLE
Event invitee options scoped by selected communities

### PR DESCRIPTION
**Pull request by lorenjohnson**
Requires: https://github.com/Hylozoic/hylo-node/pull/500

In addition to things mention in attached tickets this update includes addition of debounce for invitee search in both edit and add contexts, better visibility of "Invite" link in Post Card view, some refactoring, and somewhat more testing. 

IMPORTANT: Until we're done QA'ing this I've left the feature flag in place. So to release this to production we need to set the following ENV variable: `FEATURE_FLAG_EVENTS=on`

\* NOTE: The notifications story related to inviting has not been carefully QA'd.

### CODE REVIEW

File: src/components/PostEditor/PostEditor.js [hunk 1]
```diff
@@ -429,6 +429,9 @@ export default class PostEditor extends React.Component {
             <MemberSelector
               initialMembers={members || []}
               onChange={this.updateProjectMembers}
+              // NOTE: Turn-on to limit Project Member selection
```
Comment (tibetsprague) [inline src/components/PostEditor/PostEditor.js:432]: i say lets turn this on. i dont actually understand how event/project invites currently work with people outside the communities it is posted to. Can they see the post even though they are not in any of the communities it is posted to?

Comment (lorenjohnson) [reply src/components/PostEditor/PostEditor.js:432]: Yes, I think this was always a known gap and due to the lift required--which I did this week--we had let it slip and were planning to loop back. Now that this is done it makes sense to catch it up here. * However, the tidy part of my brain wants to push this branch/PR and then apply this to projects as a separate branch/PR (I can get that out pretty quickly on Monday).

**Comment by lorenjohnson** [comment]
We good to merge? * Once it's released to master we need to set the `FEATURE_FLAG_EVENTS=on` on hylo-evo production, but we don't want to set that until this release is on production so as to not expose the currently buggy version, if that makes sense. 

* I'm out camping and out of range from now'ish til I'm back online on Sunday night or Monday AM (my time). I'm ok with this going out now, but if anything comes-up you won't be able to find me until then probably.

**Comment by tibetsprague** [comment]
@lorenjohnson why wouldnt we just remove the need for the FEATURE_FLAG completely? i guess i wanted to turn on for projects too before we merge but otherwise yes!

**Comment by tibetsprague** [comment]
ok, decided to just merge this as is, and keep testing. and yes do a separate branch for projects

**Pull request merged**